pkg/resources/cruisecontrol/configmap.go

```diff
@@ -162,30 +159,47 @@ func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger,
 
 	capacityConfig := CapacityConfig{}
 
-	for _, broker := range kafkaCluster.Spec.Brokers {
-		brokerDisks, err := generateBrokerDisks(broker, kafkaCluster.Spec, log)
-		if err != nil {
-			return "", errors.WrapIfWithDetails(err, "could not generate broker disks config for broker", "brokerID", broker.Id)
+	brokerIdFromStatus := make([]string, 0, len(kafkaCluster.Status.BrokersState))
+	for key := range kafkaCluster.Status.BrokersState {
+		brokerIdFromStatus = append(brokerIdFromStatus, key)
+	}
+	sort.Strings(brokerIdFromStatus)
+	log.Info("String keys", "brokerIdFromStatus", brokerIdFromStatus)
+
+	for _, brokerId := range brokerIdFromStatus {
```

Comment:
Instead of collecting the broker ids into a separate array wouldn't be more straightforward something like this?
```
for _, broker := range kafkaCluster.Spec.Brokers {
  if _, ok := kafkaCluster.Status.BrokersState[strconv.Itoa(int(broker.Id))]; !ok {
    brokerCapacity = generateDefaultBrokerCapacityWithId(brokerId)
    continue
  }

....
....
}

```

Reply:
I don't think that would work right. We had the -1 indexed default config because of PR #556, where it's stated that it was added because of ungraceful broker deletions, or rather because CruiseControl might not be aware of those brokers and thus a default value was needed to deal with those. Now this caused some problems, which I'm aiming to solve here by getting rid of the -1 default config and adding a default to every broker based on state rather than spec, where they should be available even if they're not in spec. That is what the meaning of the brokerFound bool is, so that if they're found in spec we can assign them capcityconfig values from there but if they're not in there we can still assign them some default values, but on a id by id basis instead of a universal default.
What your way would achieve (if I understand correctly) is iterating through the spec and if we can't find a broker already found in the spec also in the status then we assign default values. I don't think that can happen really.